dss/events/__init__.py

```diff
@@ -54,23 +54,51 @@ def record_event_for_bundle(replica: Replica,
     """
     Build the bundle metadata document, record it into flashflood, and return it
     """
-    # TODO: Add support for unversioned tombstones
+    if not DSS_BUNDLE_KEY_REGEX.match(key):
+        raise DSSEventRecordError(f"Event recording not supported for: {key}")
     fqid = key.split("/", 1)[1]
     if flashflood_prefixes is None:
         flashflood_prefixes = replica.flashflood_prefix_write
-    metadata_document = _build_bundle_metadata_document(replica, key)
     if use_version_for_timestamp:
         _, version = fqid.split(".", 1)
         event_date = datetime_from_timestamp(version)
     else:
         event_date = datetime.utcnow()
+    metadata_document = build_bundle_metadata_document(replica, key)
     for pfx in flashflood_prefixes:
         ff = FlashFlood(resources.s3, Config.get_flashflood_bucket(), pfx)
         if not ff.event_exists(fqid):
             ff.put(json.dumps(metadata_document).encode("utf-8"), event_id=fqid, date=event_date)
     return metadata_document
 
-def _build_bundle_metadata_document(replica: Replica, key: str) -> dict:
+def delete_event_for_bundle(replica: Replica, key: str, flashflood_prefixes: typing.Tuple[str, ...]=None):
```

Comment:
Might be nice to have a comment or two on what cases each of these conditional blocks are handling